src/api/Controllers/UploadController.cs

```diff
@@ -42,6 +44,95 @@ public UploadController(BdmsContext context, ILogger<UploadController> logger, L
         this.boreholeFileCloudService = boreholeFileCloudService;
     }
 
+    /// <summary>
+    /// Receives an uploaded JSON file to import one or several <see cref="Borehole"/>(s).
+    /// </summary>
+    /// <param name="workgroupId">The <see cref="Workgroup.Id"/> of the new <see cref="Borehole"/>(s).</param>
+    /// <param name="file">The <see cref="IFormFile"/> containing the borehole JSON records that were uploaded.</param>
+    /// <returns>The number of the newly created <see cref="Borehole"/>s.</returns>
+    [HttpPost("json")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    [RequestSizeLimit(int.MaxValue)]
+    [RequestFormLimits(MultipartBodyLengthLimit = MaxFileSize)]
+    public async Task<ActionResult<int>> UploadJsonFile(int workgroupId, IFormFile file)
+    {
+        // Increase max allowed errors to be able to return more validation errors at once.
+        ModelState.MaxAllowedErrors = 1000;
+
+        if (file == null || file.Length == 0)
+        {
+            return BadRequest("No file uploaded.");
+        }
+
+        if (!FileTypeChecker.IsJson(file)) return BadRequest("Invalid file type for borehole JSON.");
+
+        try
+        {
+            using var stream = file.OpenReadStream();
+            using var reader = new StreamReader(stream);
+            var jsonContent = await reader.ReadToEndAsync().ConfigureAwait(false);
+
+            List<BoreholeImport>? boreholes;
+
+            try
+            {
+                boreholes = JsonSerializer.Deserialize<List<BoreholeImport>>(jsonContent, jsonImportOptions);
+            }
+            catch (JsonException)
+            {
+                return BadRequest("The provided file is not a array of boreholes or is not a valid JSON format.");
+            }
+
+            if (boreholes == null || boreholes.Count == 0)
+            {
+                return BadRequest("No boreholes found in file.");
+            }
+
+            for (var i = 0; i < boreholes.Count; i++)
+            {
+                boreholes[i].ImportId = i + 1;
+            }
```

Comment:
Q: Was wird hier gemacht und wieso braucht es das? Generell fehlen ein paar Codekommentare in dieser Controller-Methode.

Reply:
Habe die Validierung angepasst. Dann braucht es die Import Id beim Json Import nicht